### PR TITLE
Fixing Unit tests for OAuth and travis script

### PR DESCRIPTION
**Pull request by kchitalia**

### CODE REVIEW

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m [hunk 1]
```diff
@@ -101,6 +137,15 @@ - (void)beginTokenEndpointFlow:(SFOAuthTokenEndpointFlow)flowType {
...
     self.tokenEndpointFlowType = flowType;
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        if (self.refreshTokenFlowIsSuccessful) {
+            [self.coordinator handleTokenEndpointResponse:[self refreshTokenSuccessData]];
+        } else {
+            //[self log:SFLogLevelDebug format:@"%@ Error: response has no payload: %@", NSStringFromSelector(_cmd), ]
+        }
+    });
+    [self.coordinator performSelector:@selector(handleTokenEndpointResponse:) withObject:[self refreshTokenSuccessData] afterDelay:self.timeBeforeUserAgentCompletion];
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: This line is duplicating the functionality of the `dispatch_after` block above.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m [hunk 2]
```diff
@@ -101,6 +137,15 @@ - (void)beginTokenEndpointFlow:(SFOAuthTokenEndpointFlow)flowType {
     [self log:SFLogLevelDebug format:@"%@ called.", NSStringFromSelector(_cmd)];
     self.beginTokenEndpointFlowCalled = YES;
     self.tokenEndpointFlowType = flowType;
+    
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: You could replace the `5.0` with `self.timeBeforeTokenRefreshCompletion`, or somesuch.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m [hunk 3]
```diff
@@ -101,6 +137,15 @@ - (void)beginTokenEndpointFlow:(SFOAuthTokenEndpointFlow)flowType {
...
+        if (self.refreshTokenFlowIsSuccessful) {
+            [self.coordinator handleTokenEndpointResponse:[self refreshTokenSuccessData]];
+        } else {
+            //[self log:SFLogLevelDebug format:@"%@ Error: response has no payload: %@", NSStringFromSelector(_cmd), ]
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: And then use `[self refreshTokenErrorData]` here?

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m (hunk 2 above)
Comment (kchitalia) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: Yep. Makes sense. Will change it.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m (hunk 1 above)
Comment (kchitalia) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: So move it into a method?

Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: The `dispatch_after` block above is doing the same thing as this line.  This line can just be removed, in favor of the `dispatch_after` block doing what it's doing.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m (hunk 3 above)
Comment (kchitalia) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: Already changed it :). Coming up in my updates to this PR

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m (hunk 1 above)
Comment (kchitalia) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: Got it. I should remove it from the beginuserAgentFlow method too.

Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SFOAuthFlowAndDelegate.m]: Oh yes, didn't see it there.  Yeah, same deal.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/SalesforceOAuthUnitTests.m [hunk 4]
```diff
@@ -254,7 +255,7 @@ - (void)testTokenEncryptionDecryption
 {
     NSString *accessToken = @"gimmeAccess!";
     NSString *refreshToken = @"IWannaRefresh!";
-    SFOAuthCredentials *credentials = [[SFOAuthCredentials alloc] initWithIdentifier:kIdentifier clientId:kClientId encrypted:YES];
+    SFOAuthKeychainCredentials *credentials = [[SFOAuthKeychainCredentials alloc] initWithIdentifier:kIdentifier clientId:kClientId encrypted:YES];
```
Comment (bhariharan) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SalesforceOAuthUnitTests.m]: Why the change to use the subclass ```SFOAuthKeychainCredentials```?

Comment (kchitalia) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SalesforceOAuthUnitTests.m]: The methods we use in here are not part of the parent class. I was seeing a lot of build errors because of this. We could move the methods to the parent class, but they are specific to the behavior of the sub class.

Comment (bhariharan) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/SalesforceOAuthUnitTests.m]: :+1:

**Comment by bhariharan** [comment]
:+1:
